src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs

```diff
@@ -1435,5 +1435,42 @@ public async Task PartnerDetails_Post_AddErrorIfMoreThanTenPartners()
 
             var model = result.Model as PartnerViewModel;
         }
+
+        [Fact]
+        public async Task SoleTraderDetails_Get_ReturnsView()
+        {
+            // Act
+            var result = await controller.SoleTraderDetails() as ViewResult;
```

Comment:
@stevelee-civica There should also be a test to check that when the 

var existingTransaction = await transactionService.GetOrganisationTransactionData(User.GetAccessToken());

Call is made in the GET method that that existing model is returned to the view.

Reply:
added unit test

Reply:
updated organisation transaction service to handle SoleTraderViewModel